### PR TITLE
Remove null index values on normalized dataframe

### LINKED ISSUE #1874: normalize_dataframe can introduce a null value into the new DataFrame's index, causing Woodwork error
Woodwork 0.12.0 no longer allowing null values in index columns has caused some of our normalize dataframe tests to fail, because the intended index column contains nans, which introduces a nan into the normalized index.

We should determine what the expected behavior is when there are nans in the intended index column. We can
- Ignore the nan, and only include non-null values in the normalized dataframe
- Not allow this situation since nans aren't allowed in indices and raise an error

### PR DESCRIPTION
**Pull request by rwedge**
Fixes #1874

This PR doesn't address #1680 as with the NA index value removed from the dataframe at normalization, there's no NA-indexed data to group on, even if we enabled the NA GROUP

### CODE REVIEW

**Comment by tamargrey** [comment]
We can probably close #1873 now @dvreed77

File: featuretools/entityset/entityset.py [hunk 1]
```diff
@@ -803,6 +803,7 @@ def normalize_dataframe(self, base_dataframe_name, new_dataframe_name, index,
 
         new_dataframe2 = new_dataframe. \
             drop_duplicates(index, keep='first')[selected_columns]
+        new_dataframe2 = new_dataframe2.dropna(subset=[index])
```
Comment (rwedge) [inline featuretools/entityset/entityset.py:806]: doing the drop here might cause a scenario where `make_secondary_time_index` logic reintroduces null index values but I'm not positive

**Comment by rwedge** [comment]
@alteryx/ml-tools, any ideas on why upgrading to WW 0.12.0 would impact this test?
https://github.com/alteryx/featuretools/blob/79eea5fd3db16d4059141af77cd69f5864f20f14/featuretools/tests/entityset_tests/test_es.py#L2175-L2180

**Review by tamargrey — APPROVED** [review]
lgtm assuming tests pass!

**Pull request merged**